examples/provider/bid/casale/headertag-ex1.html

```diff
@@ -0,0 +1,442 @@
+            var v = '9';
+            var x3 = '1';
+            var scriptSrc = requestUrl;
+            scriptSrc += '/headertag?v=' + v + '&x3=' + x3;
+            if (pf.util.asType(cygnus_index_args.parseFn) === 'function') {
+                scriptSrc += '&fn=' + 'cygnus_index_args.parseFn';
+            }
+            scriptSrc += '&s=' + this.indexArgs.siteID + '&r=' + request;
+            return scriptSrc;
+        };
+
+        try {
+            if (cygnus_index_args.hasOwnProperty('slots') === false) {
+                return;
+            }
+            if (typeof _IndexRequestData === 'undefined') {
+                _IndexRequestData = {};
+                _IndexRequestData.impIDToSlotID = {};
+                _IndexRequestData.reqOptions = {};
+            }
+
+            var req = new OpenRTBRequest();
+            if (cygnus_index_args.url && pf.util.asType(cygnus_index_args.url) === 'string' && !cygnus_index_args.url.match(/^\s*$/)) {
+                    this.sitePage = cygnus_index_args.url;
+            }
+
+            _IndexRequestData.impIDToSlotID[req.requestID] = {};
+            _IndexRequestData.reqOptions[req.requestID] = {};
+            var slotDef, impID;
+
+            for (var i = 0; i < cygnus_index_args.slots.length; i++) {
+                slotDef = cygnus_index_args.slots[i];
+
+                impID = req.addImpression(slotDef.width, slotDef.height, slotDef.bidfloor, slotDef.bidfloorcur, slotDef.id, slotDef.siteID);
+                if (impID) {
+                    _IndexRequestData.impIDToSlotID[req.requestID][impID] = String(slotDef.id);
+                }
+            }
+
+            if (pf.util.asType(cygnus_index_args.targetMode) === 'number') {
+                _IndexRequestData.reqOptions[req.requestID].targetMode = cygnus_index_args.targetMode;
+            }
+
+            if (pf.util.asType(cygnus_index_args.callback) === 'function') {
+                _IndexRequestData.reqOptions[req.requestID].callback = cygnus_index_args.callback;
+            }
+
+            return req.buildRequest();
+        } catch (e) {
+        }
+    };
+    // End Casale specific code
+    // Add our adUnits/Slots to PubFood
+    pf.addSlot({
+        name: '/2476204/multi-size',
+        elementId: 'div-multi-size',
+        sizes: [
+            [300, 250]
+        ],
+        bidProviders: [
+                'casale'
+        ]
+    });
+    pf.addSlot({
+        name: '/2476204/leaderboard',
+        elementId: 'div-leaderboard',
+        sizes: [
+            [728, 90]
+        ],
+        bidProviders: [
+            'casale'
+        ]
+    });
+    // Add a bid provider to PubFood for Casale / Index Exchange
+    indexExchange = pf.addBidProvider({
+        name: 'casale',
+        init: function(slots, pushBid, done) {
+            console.log('bidProvider.init.');
+            var fancyWorkAround = indexExchange.parseResponseFactory(slots, pushBid, done);
```

Comment:
This technique creates a closure for the function argument values to be available when processing is required.

`fancyWorkAround` should be renamed as it could be interpreted incorrectly. Creating closures is standard procedure in some JS contexts.